providers/base/bin/camera_test.py

```diff
@@ -505,30 +524,47 @@ def _capture_image_gstreamer(self, filename, width, height, pixelformat):
             pipeline.add(bayer2rgb)
             caps.link(bayer2rgb)
             rgb_capture = bayer2rgb
-
         else:
+            mime_type = "video/x-raw"
+            if pixelformat == "MJPG":
+                mime_type = "image/jpeg"
             caps.set_property(
                 "caps",
                 self.Gst.Caps.from_string(
-                    "video/x-raw,width={},height={}".format(width, height)
+                    "{},width={},height={}".format(mime_type, width, height)
                 ),
             )
             pipeline.add(caps)
             rgb_capture = caps
 
-        # Add encoder
-        encoder = self.Gst.ElementFactory.make("jpegenc", "encoder")
-        pipeline.add(encoder)
+        # Add valve, note that we can't rely on the default value of drop
+        # it varies by installations
+        valve = self.Gst.ElementFactory.make("valve", "photo-valve")
+        assert valve, "Valve element could not be created"
+        valve.set_property("drop", True)
+        pipeline.add(valve)
+
+        # Link elements
+        source.link(caps)
+        rgb_capture.link(valve)
 
-        # Add sink
-        sink = self.Gst.ElementFactory.make("filesink", "sink")
+        # Add sink, use multifilesink to avoid large jpeg files.
+        # filesink does not stop writing after it receives the 1st buffer
+        sink = self.Gst.ElementFactory.make("multifilesink", "sink")
         sink.set_property("location", filename)
         pipeline.add(sink)
 
-        # Link elements
-        source.link(caps)
-        rgb_capture.link(encoder)
-        encoder.link(sink)
+        if pixelformat == "MJPG":
+            valve.link(sink)
+
+        else:
+            # Add encoder
+            encoder = self.Gst.ElementFactory.make("jpegenc", "encoder")
+            pipeline.add(encoder)
+            valve.link(encoder)
+            encoder.link(sink)
+
+        # source ! rgbcapture ! valve ! encoder ! filesink
```

Comment:
I like this pipeline explanation, just add the right one for each one of the cases:
```suggestion
        if pixelformat == "MJPG":
            # source ! rgbcapture ! valve ! filesink
            valve.link(sink)

        else:
            # source ! rgbcapture ! valve ! encoder ! filesink
            # Add encoder
            encoder = self.Gst.ElementFactory.make("jpegenc", "encoder")
            pipeline.add(encoder)
            valve.link(encoder)
            encoder.link(sink)
```